lightning/src/ln/channel.rs

```diff
@@ -301,6 +301,21 @@ pub struct CounterpartyForwardingInfo {
 	pub cltv_expiry_delta: u16,
 }
 
+/// When a node increases the feerate on an outbound channel, it can cause the channel to get
+/// "stuck" if the other side holds most of the funds, and suddenly is unable to send an HTLC as it
+/// would effectively debit the channel initiator to pay for the new HTLC's fee. If the initiator's
+/// balance is right at or near the channel reserve, neither side will be able to send an HTLC.
+/// Thus, before sending an HTLC when we are the initiator, we check that the feerate can increase
+/// by this multiple without hitting this case, before sending.
+/// This multiple is effectively the maximum feerate "jump" we expect until more HTLCs flow over
```

Comment:
This is the issue solved by https://github.com/lightningnetwork/lightning-rfc/pull/740 right ? 

IIRC, the new reserve is to "cover" the cost of new HTLC flowing, though it can also serve as a `update_fee` reserve but maybe in that case we should bump it a bit more compared to spec recommendations ? 

Odds of channel stuck are going to decrease with only the commitment transaction size decreasing or the initiator balance increasing (though not in case of mempools emptying as we base the upper bound on CHAN_STUCK_FEE_INCREASE_MULTIPLE * 2 in the newer `ChannelManager::update_channel_fee` ?

Reply:
Yes, that's the relevant issue. Note that there is no way to "properly" protect ourselves because its not enough to just bound single fee increases by some constant, we have to bound the total fee increase between now and when we next send an HTLC to the constant. I agree we should increase this beyond 2x, but its out of scope of this PR and requires fixing a number of tests.